Sources/Player/OfflineEngine/Internal/Storage/OfflineEntry.swift

```diff
@@ -0,0 +1,104 @@
+import AVFoundation
+import Foundation
+
+struct OfflineEntry: Codable {
+	let productId: String
+	let productType: ProductType
+	let assetPresentation: AssetPresentation
+	let audioMode: AudioMode?
+	let audioQuality: AudioQuality?
+	let audioCodec: AudioCodec?
+	let audioSampleRate: Int?
+	let audioBitDepth: Int?
+	let videoQuality: VideoQuality?
+	var revalidateAt: UInt64?
+	var expiry: UInt64?
+	let mediaType: String?
+	let albumReplayGain: Float?
+	let albumPeakAmplitude: Float?
+	let trackReplayGain: Float?
+	let trackPeakAmplitude: Float?
+	var mediaUrl: URL?
+	var licenseUrl: URL?
+
+	var state: OfflineState {
```

Comment:
[question] are these values same as in other platforms or do we keep different state for each platform?
[question 2] I see same `OFFLINED_BUT_NOT_VALID` and one `OFFLINED_AND_VALID`, so just checking indeed we want the same one in all conditions, or maybe we want to be more specific with the not valid condition, I think it can be good for future logging so we can more easily diagnose if it comes to it.

Reply:
These are not coordinated with other platforms.

I could see us adding an extra state for `OFFLINED_BUT_EXPIRED` and `OFFLINED_BUT_NO_LICENSE`

Reply:
Ok, I lied 😅 

This is part of the spec, so adding more would need us to sync with Android.
https://github.com/tidal-music/tidal-sdk/blob/main/Player.md#offlinestate

We could have an `InternalOfflineState` with more granularity, and map to OfflineState in the public API.

Reply:
Added `InternalOfflineState` ✅

Reply:
Oh nice, yes, maybe it doesn't make sense to expose all. Well, we can always get rid of it it proves to not be useful.